docs/contributing/MODEL_MANAGER.md

```diff
@@ -0,0 +1,1214 @@
+# Introduction to the Model Manager V2
+
+The Model Manager is responsible for organizing the various machine
+learning models used by InvokeAI. It consists of a series of
+interdependent services that together handle the full lifecycle of a
+model. These are the:
```

Comment:
So far, I'm really liking the improved modularity of this design.

One thing I'm wondering is if we should more clearly define the model API that invocations are intended to interact with. Specifically, these services expose a bunch of useful endpoints, but it seems like only a small subset should probably be used by invocations.

Separately, do we have a definition somewhere of what a "Service" is intended to be? There is a lot more interdependency that I would have expected between these services. I think those interdependencies are necessary - I'm more so questioning if they should all be defined on equal footing as "Services".

Reply:
Sounds good. We can address the context being too powerful separately from this model manager work. I'll find some time to review https://github.com/invoke-ai/InvokeAI/issues/4887 and https://github.com/invoke-ai/InvokeAI/pull/4910.

And, thanks for the background on Services.

Reply:
Good points. From the point of view of the MM refactor, is this actionable?

What I'd like guidance on is how to split code among `invokeai/backend/model_manager` and `invokeai/app/services`. Right now there's a bit of mirroring going on in which code that goes into `backend` is anything that doesn't need to know anything about the invocation context and event bus, while code in `services` builds on top of the backend code using either composition or inheritance. Is this the best thing to do or should everything get stuffed into `services`?